adsmp/app.py

```diff
@@ -150,12 +152,17 @@ def update_storage(self, bibcode, type, payload):
             else:
                 raise Exception('Unknown type: %s' % type)
             session.add(ChangeLog(key=bibcode, type=type, oldvalue=oldval))
-            
+
             r.updated = now
             out = r.toJSON()
-            session.commit()
-            return out
-
+            try:
+                session.commit()
+                return out
+            except exc.IntegrityError:
+                msg = 'error in app.update_storage while updating database for bibcode {}, type {}'.format(bibcode, type)
+                self.logger.exception(msg)
+                session.rollback()
+                return msg
```

Comment:
I did not have an issue returning `None` but about what was done with that value in places like https://github.com/adsabs/ADSMasterPipeline/blob/master/adsmp/tasks.py#L63. Returning a string with success and failure makes it harder to know if the method succeeded from outside.